app/features/settings/contacts.tsx

```diff
@@ -1,7 +1,18 @@
+import { useState } from 'react';
 import { PageContent } from '~/components/page';
+import { AddContactDrawer, ContactsList } from '~/features/contacts';
 import { SettingsViewHeader } from '~/features/settings/ui/settings-view-header';
+import { SearchBar } from '../../components/search-bar';
+import { useContacts } from '../contacts/contact-hooks';
 
 export default function Contacts() {
+  const contacts = useContacts();
```

Comment:
wouldn't proper way be to have this hook handle filtering and then you can fetch from the db only what you need? then the hook could also handle pagination etc

Reply:
I think the ideal solution would be to have getAll method on repository not to be get all, but paginated with filtering. It could be called `query` or `getPage` or something like that. The method would do the filtering on the db and then pagination and sorting is also applied on the db on the filtered records. Then when you scroll to the bottom you get the next page. 
If we don't think users will have many contacts for now and pagination is overkill at this point, I would still do it the way I described but just without the pagination (so filtering also done on db)

Reply:
Yeah if we will pull everything in memory then we can leave it there. I was thinking maybe we just return all the search results without pagination but not everything.

Let's just then pull everything for now and limit the number of contacts one user can have with db constraint